bugbug/bug_features.py

```diff
@@ -232,12 +232,18 @@ def __call__(self, bug, **kwargs):
 
 class components_touched(object):
     def __call__(self, bug, **kwargs):
-        return (ct for commit in bug['commits'] for ct in commit['components'])
+        components = set()
+        for commit in bug['commits']:
+            components.update(commit['components'])
+        return list(components)
```

Comment:
Can you check if this feature works also without converting the set into a list?

Reply:
Should I test this by running `bug.py` model?

Reply:
You can choose any model, just try to add it and see if it works without the `list` conversion. If it does, then just remove the conversion.

Reply:
Running the script raises a KeyError for `bug['commits']`.

And there are no `commits` in `bugs.json` even after updating the database. Do I need to do something else to get the commits?

Reply:
You just need to pass commit_data=True, https://github.com/mozilla/bugbug/blob/6950562854a544d6f6929c487a61d65fa54d59ea/bugbug/bug_features.py#L291

Reply:
It doesn't work without `list` conversion. So can we instead use a set comprehension first, convert it into a list and return?

Reply:
Sure, it was just a test to see if we could avoid the conversion. If we can't avoid it, then we have to do it ;)